### DISCUSSION TITLE
Quit apps instantly to avoid delayed app switching

### DISCUSSION THREAD

**Discussion by lawand-dot-io** [aleppo-io/last-window-quits#20] [Feature requests]
Feedback I recieved by email:

> In the Advanced tab, under “App quit timeout”, is it possible to decrease the time below 1 second?
> Currently the slider is capped at 1s minimum, but some MacOS apps close instantly, by design. 
> 
> For example, Final Cut Pro, this app closes instantly when using the close button, with no time out.
> Can we make the same effect for Last Window Quits app? To enable an option where we can manually type 0 (zero seconds), or instant, or 0.2 seconds.
> 
> This would help a lot, because when using Stage Manager and closing an app, after 1-2 seconds the app closes and then it automatically switches the screen to another app, which makes it confusing. This is normal MacOS behavior, but it would be much easier to have near-instant app close option.

**Comment by lawand-dot-io** [comment]
I am aware of this issue (apps being switching 1-2 seconds after the quit) but the problem is I make the timer zero is that this breaks many apps. For example, Microsoft Teams shows a splash window then closes that window and then opens the main window. This will be detected by "Last Window Quits" as the app's last window being closed and will quit Teams, which means that that app can't be useds. I will continue to think about a better solution than the 1-2 second timer.

**Comment by lawand-dot-io** [comment]
Further feedback:

> There are a few different options here:
> - you could add another option in the slider for 0.5 sec, for Instant.
> - if this affects some apps, users can simply just add those apps to the black list.
> - an additional information panel can be aded in you app, to clearly instruct users that lowering the timeout limit to 0 might break some apps
> - this could be added in an additional “Advanced settings” where only tech people would go to make those changes.

**Comment by afrey15** [comment]
Another option: a whitelist for apps to have 0 second delay. There are many apps that do not break if closed without a delay and the more of these that can be set, the less annoying the issue is with stage manager. You can warn in the whitelist that certain apps might break if added.
